src/main/game/gameboard/Tile.java

```diff
@@ -2,203 +2,108 @@
-//        this.location = location;
-//        this.setOwnerID(-1);
-//    }
-//
-//    // Get Tile Location
-//    public Location getLocation() {
-//        return location;
-//    }
-//
-//    // Add unit to Tile
-//    public void addUnit(Unit unit) {
-//        units.add(unit);
-//        unit.setLocation(this.location);
-//        containsUnit = true;
-//    }
-//
-//    // Test if terrain is impassable
-//    public boolean isImpassable() {
-//        return (Terrain == TerrainEnum.INVISIBLE || Terrain == TerrainEnum.WATER) ? true : false;
-//    }
-//
-//    // Get Tile terrain type
-//    public TerrainEnum getTileType() {
-//        return Terrain;
-//    }
-//
-//    // Remove unit from tile by ID
-//    public void removeUnit(int unitID) {
-//        for (int i = 0; i < units.size(); i++) {
-//            if (units.get(i).getUnitID() == unitID) {
-//                units.remove(i);
-//            }
-//        }
-//
-//        if (units.isEmpty()) {
-//            containsUnit = false;
-//        }
-//    }
-//
-//    public void removeAllUnits() {
-//        this.units.clear();
-//    }
-//
-//    public ArrayList<RallyPoint> getRallyPoints() {
-//        return rallyPoints;
-//    }
-//
-//    public void addRallyPoint(RallyPoint rp) {
-//        rallyPoints.add(rp);
-//        rp.setLocation(location);
-//        containsRallyPoint = true;
-//    }
-//
-//    public void removeRallyPoint(RallyPoint rp) {
-//        for (int i = 0; i < rallyPoints.size(); i++) {
-//            if (rallyPoints.get(i).getRallyID() == rp.getRallyID()) {
-//                rallyPoints.remove(i);
-//                break;
-//            }
-//        }
-//        if (rallyPoints.isEmpty()) {
-//            containsRallyPoint = false;
-//        }
-//    }
-//
-//
-//    // Get all units from Tile
-//    public ArrayList<Unit> getUnits() {
-//        return units;
-//    }
-//
-//    // Add army to Tile
+    private Structure structure;
+    private Location location;
+    private int ownerId;
+
+    Tile(TerrainEnum tileType, Location location) {
+        this.Terrain = tileType;
+        this.units = new ArrayList<Unit>();
+        //this.armies = new ArrayList<Army>();
+        //this.rallyPoints = new ArrayList<RallyPoint>();
+        this.containsStructure = false;
+        this.containsRallyPoint = false;
+        this.containsUnit = false;
+        this.containsArmy = false;
+        this.structure = null;
+        this.location = location;
+        this.setOwnerId(-1);
+    }
+
+    public void setOwnerId(int ownerId){
+        this.ownerId = ownerId;
+    }
+
+    public int getOwner() {
+        return this.ownerId;
+    }
+
+    public Location getLocation(){return location;}
+
+    public void addUnit(Unit unit){
+        units.add(unit);
+        //unit.setLocation(this.location);
+    }
+
+    public void addStructure(Structure structure){
+        if(containsStructure()==false){
+            this.structure = structure;
```

Comment:
If you look at the containsStructure(), the boolean is determined base on if the structure == null, thus, once the structure is created the method will automatically return false;